src/Microsoft.SqlTools.ServiceLayer/QueryExecution/Batch.cs

```diff
@@ -353,7 +353,7 @@ private bool IsExecutingMultipleTimes()
                 dbCommand.CommandText = BatchText;
                 dbCommand.CommandType = CommandType.Text;
                 dbCommand.CommandTimeout = 0;
-                executionStartTime = DateTime.Now;
+                executionStartTime = this.BatchExecutionCount > 1 ? executionStartTime : DateTime.Now;
```

Comment:
I fixed this by just moving the assignment of this property. Take a look at mine. Your's has tests so we should take yours, but I think my fix was cleaner.

Reply:
Sorry, I didn't realize you already had a PR out for this.